src/Approvals/SFA.DAS.Approvals/Application/Cohorts/Queries/GetHasDeclaredStandards/GetHasDeclaredStandardsQueryHandler.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using SFA.DAS.Approvals.InnerApi.Requests;
+using SFA.DAS.Approvals.InnerApi.Responses;
+using SFA.DAS.Approvals.Services;
+using SFA.DAS.SharedOuterApi.Configuration;
+using SFA.DAS.SharedOuterApi.Extensions;
+using SFA.DAS.SharedOuterApi.Interfaces;
+using Party = SFA.DAS.Approvals.Application.Shared.Enums.Party;
+
+namespace SFA.DAS.Approvals.Application.Cohorts.Queries.GetHasDeclaredStandards
+{
+    public class GetHasDeclaredStandardsQueryHandler : IRequestHandler<GetHasDeclaredStandardsQuery, GetHasDeclaredStandardsQueryResult>
+    {
+        private readonly ICommitmentsV2ApiClient<CommitmentsV2ApiConfiguration> _apiClient;
+        private readonly IProviderStandardsService _providerStandardsService;
+        private readonly ServiceParameters _serviceParameters;
+
+        public GetHasDeclaredStandardsQueryHandler(
+            ICommitmentsV2ApiClient<CommitmentsV2ApiConfiguration> apiClient,
+            IProviderStandardsService providerStandardsService,
+            ServiceParameters serviceParameters)
+        {
+            _apiClient = apiClient;
+            _providerStandardsService = providerStandardsService;
+            _serviceParameters = serviceParameters;
+        }
+
+        public async Task<GetHasDeclaredStandardsQueryResult> Handle(GetHasDeclaredStandardsQuery request, CancellationToken cancellationToken)
+        {
+            var providerId = _serviceParameters.CallingParty == Party.Employer
```

Comment:
The calling party shouldn't even be `Employer`, and I think we could enforce that here. As mentioned before, ProviderId need not be in the query as it can be read from the ServiceParameters.

Reply:
Done